cli/flox/src/commands/environment.rs

```diff
@@ -783,8 +783,47 @@ pub struct Push {
 }
 
 impl Push {
-    pub async fn handle(self, flox: Flox) -> Result<()> {
+    pub async fn handle(self, mut flox: Flox) -> Result<()> {
         subcommand_metric!("push");
+
+        if flox.floxhub_token.is_none() {
+            let options = indoc! {"
+                * login to floxhub with 'flox auth login',
+                * set the 'floxhub_token' field to '<your token>' in your config
+                * set the '$FLOX_FLOXHUB_TOKEN=<your_token>' environment variable."};
+
+            let dialog_help = formatdoc! {"
+                Alternatively you can also
+                {options}"};
+
+            let dialog = Dialog {
+                message: "You are not logged in to floxhub. Would you like to login now?",
```

Comment:
suggestion: as logging in is the only possible solution, we don't need to confirm logging in. See style guide

Reply:
It's still requiring user interaction (the login) asking if users want to do that seems reasonable -- i understand the style guide as avoinding interaction if an automated thing can be done

Reply:
```
Example: “`flox push` without auth should automatically flox auth login and print a sentence: “Auth information not found. Redirecting to ‘flox auth login’.”
```

Reply:
hmm

Reply:
ok it does say redirect automatically 🤔
Changed..